actors/builtin/miner/miner_actor.go

```diff
@@ -1610,6 +1611,24 @@ func handleProvingDeadline(rt Runtime) {
 				st.ProvingPeriodStart = st.ProvingPeriodStart + WPoStProvingPeriod
 			}
 		}
+
+		// expire pre-committed sectors
+		expiryQ, err := LoadBitfieldQueue(store, st.PreCommittedSectorsExpiry, quant)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load sector expiry queue")
+
+		bf, modified, err := expiryQ.PopUntil(currEpoch)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to pop expired sectors")
+
+		if modified {
+			st.PreCommittedSectorsExpiry, err = expiryQ.Root()
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to save expiry queue")
+		}
+
+		if bf != nil {
```

Comment:
this should never return nil, unless it returns an error.

Reply:
Have removed this check and added an Assert for this.